ext/pdo/pdo_stmt.c

```diff
@@ -1129,28 +1150,29 @@ static int pdo_stmt_verify_mode(pdo_stmt_t *stmt, zend_long mode, int fetch_all)
 	switch(mode) {
 		case PDO_FETCH_FUNC:
 			if (!fetch_all) {
-				pdo_raise_impl_error(stmt->dbh, stmt, "HY000", "PDO::FETCH_FUNC is only allowed in PDOStatement::fetchAll()");
+				zend_argument_value_error(mode_arg_num, "can only use PDO::FETCH_FUNC in PDOStatement::fetchAll()");
```

Comment:
So this will print something like `Argument #n ($fetch_style) can only use PDO::FETCH_FUNC in PDOStatement::fetchAll()` which doesn't really make sense.